lib/geometry/polygon.nit

```diff
@@ -0,0 +1,428 @@
+# This file is part of NIT (http://www.nitlanguage.org).
+#
+# Copyright 2014 Romain Chanoir <romain.chanoir@viacesi.fr>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Convex Polygons manipulations
+module polygon
+
+import points_and_lines
+
+
+# Convex Polygon class
+class ConvexPolygon
+
+	# Vertices of this polygon
+	var points = new Array[Point[Float]]
+
+	# Init this polygon with a list of at least 3 vertices
+	init with_vertices(pts: Array[Point[Float]]) do
+		assert pts.length >= 3
+		points.add_all(pts)
+	end
+
+	# Get an array of x coordinates of the vertices
+	private fun x_coordinates: Array[Float] do
+		var x = new Array[Float]
+		for p in points do
+			x.add(p.x)
+		end
+		return x
+	end
+
+	# Get an array of y coordinates of the vertices
+	private fun y_coordinates: Array[Float] do
+		var y = new Array[Float]
+		for p in points do
+			y.add(p.y)
+		end
+		return y
+	end
+
+	# Get a matrice containing the coordinates of the vertices
+	private fun vertices: Array[Array[Float]] do
+		var vertices = new Array[Array[Float]]
+		for i in [0..points.length[ do
+			var temp = new Array[Float]
+			temp.add(points[i].x)
+			temp.add(points[i].y)
+			vertices.add(temp)
+		end
+		return vertices
+	end
+
+	# Get axes for the intersection test
+	private fun get_axes: Array[Point[Float]] do
+		var axes = new Array[Point[Float]]
+		for i in [0..points.length[ do
+			var p1 = new Point[Float](points[i].x, points[i].y)
+			var p2 = new Point[Float](points[(i+1) % points.length].x, points[(i+1) % points.length].y)
+			var edge = new Point[Float](p1.x - p2.x, p1.y - p2.y)
+			var normal = new Point[Float](-edge.y, edge.x)
+			axes[i] = normal
+		end
+		return axes
+	end
+
+	# Sort the vertices in counter clockwise order
+	# ~~~
```

Comment:
I don't know if this is requirement, but as a personal preference, I like having a blank line between the nitunits and the synopsis